### PR TITLE
Hugepage config cleanup

### PR DESCRIPTION
**Pull request by cmcfarlen**
This removes the following settings:

  - proxy.config.cache.dir.enable_hugepages
  - proxy.config.allocator.iobuf_use_hugepages

Instead, these features will just use the `proxy.config.allocator.hugepages` setting.  This is to reduce the config required to get huge pages going.

This PR also adds back in the hopefully more user friendly parsing of the iobuffer buckets.  This is related because this was part of the prior hugepage work but was left out because libswoc was not linked to the binaries in the build system.

### CODE REVIEW

File: iocore/eventsystem/IOBuffer.cc [hunk 1]
```diff
@@ -73,6 +74,80 @@ init_buffer_allocators(int iobuffer_advice)
...
+      }
+
+      // Check if n goes out of bounds
+      if (n >= DEFAULT_BUFFER_SIZES) {
+        Error("Invalid IO buffer chunk sizes string");
+        return false;
+      }
+
+      auto x = ts::svto_radix<10>(token);
+      if (token.empty() && x <= std::numeric_limits<int>::max()) {
```
Comment (brbzull0) [inline iocore/eventsystem/IOBuffer.cc:138]: This may break in ubuntu  ` x <= std::numeric_limits<int>::max()`, you fixed this already ;)

Comment (cmcfarlen) [reply iocore/eventsystem/IOBuffer.cc:138]: oops, unfixed!  Thanks!

Comment (cmcfarlen) [reply iocore/eventsystem/IOBuffer.cc:138]: I do wish `svto_radix` returned `std::optional` or something!

**Comment by cmcfarlen** [comment]
[approve ci autest]

File: iocore/eventsystem/EventSystem.cc [hunk 2]
```diff
@@ -45,24 +45,12 @@ ink_event_system_init(ts::ModuleVersion v)
...
-      if (token.empty() && x != std::numeric_limits<decltype(x)>::max()) {
-        chunk_sizes[n++] = x;
-      } else {
-        break;
-      }
-    }
-    ats_free(chunk_sizes_string);
+  if (chunk_sizes_string && !parse_buffer_chunk_sizes(chunk_sizes_string, chunk_sizes)) {
+    // If we can't parse the string then we can't be sure of the chunk sizes so just exit
+    ::exit(1);
```
Comment (ywkaras) [inline iocore/eventsystem/EventSystem.cc:50]: Why aren't you using `Fatal()` from `Diags.h`?

Comment (cmcfarlen) [reply iocore/eventsystem/EventSystem.cc]: Great question! Fixed

**Review by ywkaras — APPROVED** [review]

**Pull request merged**